api/v1beta2/kustomization_types.go

```diff
@@ -0,0 +1,361 @@
+/*
+Copyright 2021 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	"time"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/fluxcd/pkg/apis/kustomize"
+	"github.com/fluxcd/pkg/apis/meta"
+	"github.com/fluxcd/pkg/runtime/dependency"
+)
+
+const (
+	KustomizationController   = "kustomize-controller"
+	KustomizationKind         = "Kustomization"
+	KustomizationFinalizer    = "finalizers.fluxcd.io"
```

Comment:
We have had a discussion about this in the past, but I think it would be wise if we would make the finalizers domain specific. Reason for this is that the pattern of using various finalizers to control the garbage collection process of an object if multiple reconcilers are dealing with an object has become more common.

I think your objection in the past was that removing "all Flux reconcilers" in cases where they get stuck would become more difficult, but if we would nest all under a sub(domain)name, you could still loop over all that match that suffix. E.g. `applier.finalizers.fluxcd.io`, `source.finalizers.fluxcd.io`.

Reply:
If we change this now, wouldn't result in orphan resources after a Flux upgrade? I would introduce this change into a followup PR.

Reply:
Follow up PR (e.g. bundled with the other refactoring efforts) is fine with me, just wanted to have it noted given it showed up again :-)